loadtest/control/simulcontroller/actions.go

```diff
@@ -1838,3 +1838,87 @@ func (c *SimulController) createPostReminder(u user.User) control.UserActionResp
 
 	return control.UserActionResponse{Info: fmt.Sprintf("created post reminder, id %s", post.Id)}
 }
+
+func (c *SimulController) reconnectWebSocket(u user.User) control.UserActionResponse {
+	team, err := c.user.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team != nil {
+		// The webapp makes these requests only if the team
+		// is set in the state.
+
+		if _, err := u.GetChannelsForTeamForUser(team.Id, u.Store().Id(), false); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		if err := u.GetChannelMembersForUser(u.Store().Id(), team.Id); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		// getPostsForChannel(?since=mostRecentPost.CreateAt in CurrentChannel)
+		crtEnabled, resp := control.CollapsedThreadsEnabled(u)
+		if resp.Err != nil {
+			return resp
+		}
+		currentCh, err := u.Store().CurrentChannel()
+		if errors.Is(err, memstore.ErrChannelNotFound) {
+			return control.UserActionResponse{Info: "current channel is not set"}
+		} else if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		posts, err := u.Store().ChannelPostsSorted(currentCh.Id, true)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		latestCreateAt := posts[len(posts)-1].CreateAt
+		if _, err := u.GetPostsSince(currentCh.Id, latestCreateAt, crtEnabled); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		// unique user ids from all posts in current Channel + users from DM prefs
+		userIds := make(map[string]bool)
+		for _, p := range posts {
+			userIds[p.UserId] = true
+		}
+		prefs, err := u.Store().Preferences()
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		for _, p := range prefs {
+			switch {
+			case p.Category == model.PreferenceCategoryDirectChannelShow:
+				userIds[p.Name] = true
```

Comment:
Should this be UserId?
```suggestion
				userIds[p.UserId] = true
```

Reply:
Ah no, UserId is the user to whom the preference belongs to. What we want are the userIDs which are in the DM channel category to show.

Reply:
Aaaah, got it, thanks!